pkg/reader/ccip.go

```diff
@@ -957,6 +953,60 @@ func (r *ccipChainReader) getOnRampDynamicConfigs(
 	return result, nil
 }
 
+type onRampDestChainConfig struct {
+	SequenceNumber   uint64 `json:"sequenceNumber"`
+	AllowListEnabled bool   `json:"allowListEnabled"`
+	Router           []byte `json:"router"`
+}
+
+//nolint:dupl // It's not quite duplicate code...
```

Comment:
I wonder if there's some generics that can be extracted. e.g:

```golang
func getOnRampConfigs[ConfigType any](
  ctx context.Context,
  srcChains []cciptypes.ChainSelector,
  methodName string,
) (map[cciptypes.ChainSelector]ConfigType, error) {
  // initial validation code ...

  result := make(map[cciptypes.ChainSelector]ConfigType)
  mu := new(sync.Mutex)
  eg := new(errgroup.Group)
  for _, chainSel := range srcChains {
    // dest chain check

    chainSel := chainSel
    eg.Go(func() error {
      var resp ConfigType
      err := r.contractReaders[chainSel].ExtendedGetLatestValue(
        ctx, consts.ContractNameOnRamp, methodName, primitives.Unconfirmed, map[string]any{}, &resp)
// .. rest of code
}
```

You might be able to make this more general, e.g also pass in the args for the call, but for now this might be enough, could reduce some duplication.

Reply:
The main problem is that there are more than just these 2 -- I just did a better job of copy/pasting here and the linter caught it :sweat_smile: 

There's already a followup ticket for switching these to using the Batch API. I updated the description to mention the refactoring as well: https://smartcontract-it.atlassian.net/browse/CCIP-3129